### PR TITLE
*: migrate test-infra to testify for util/printer.go

### LINKED ISSUE #26982: migrate test-infra to testify for `util/printer` pkg

### PR DESCRIPTION
**Pull request by jyz0309**
Signed-off-by: jyz0309 <45495947@qq.com>

### What problem does this PR solve?

Issue Number: close #26982  

Problem Summary:
migrate test-infra to testify for util/printer.go
### What is changed and how it works?

What's Changed:
migrate test-infra to testify for `util/printer.go`
How it Works:

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by xhebox — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
/cc @mmyj @tiancaiamao 

@jyz0309 release note is for user facing changes. Code refactor is almost unrelated. Update as none release note.

**Review by mmyj — APPROVED** [review]
lgtm

File: util/printer/printer_test.go [hunk 1]
```diff
@@ -16,27 +16,16 @@ package printer
...
-func (s *testPrinterSuite) TestPrintResult(c *C) {
-	defer testleak.AfterTest(c)()
+func TestPrintResult(t *testing.T) {
+	t.Parallel()
 	cols := []string{"col1", "col2", "col3"}
 	datas := [][]string{{"11"}, {"21", "22", "23"}}
 	result, ok := GetPrintResult(cols, datas)
-	c.Assert(ok, IsFalse)
-	c.Assert(result, Equals, "")
+	assert.False(t, ok)
```
Comment (mmyj) [inline util/printer/printer_test.go:27]: `assert.False` -> `require.False`?

**Review by mmyj — CHANGES_REQUESTED** [review]
What's difference between `assert` and `require`.

File: util/printer/printer_test.go (hunk 1 above)
Comment (mmyj) [reply util/printer/printer_test.go:27]: We need using `require` instead of `assert`. @jyz0309

Comment (xhebox) [reply util/printer/printer_test.go:27]: tisonkun had used `assert` in the early migration PRs for most assertions but important assertions. And I am not so against with it.

Comment (jyz0309) [reply util/printer/printer_test.go]: Done. PTAL~

Comment (tisonkun) [reply util/printer/printer_test.go]: @mmyj @xhebox now I tend to use `require` which keeps the same semantic with `c.Assert`. Early migration is lack of best practice and I think I didn't get it better then. It is not a tech blocker if there is no effective difference between calling `Fail` and calling `FailNow`, but I tend to fail test fast.

File: util/printer/printer_test.go [hunk 2]
```diff
@@ -14,29 +14,17 @@
 package printer
 
 import (
+	"github.com/stretchr/testify/require"
```
Comment (mmyj) [inline util/printer/printer_test.go:17]: We should use `goimports` to format it.

Comment (jyz0309) [reply util/printer/printer_test.go]: Done. PTAL

**Review by mmyj — APPROVED** [review]
lgtm

**Comment by mmyj** [comment]
PTAL @xhebox, need one more lgtm.

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**